src/main.lua

```diff
@@ -1108,15 +1112,22 @@ addEventingHandler(ActionMap.TokenCost, utils.hasMatchingTag("Action", ActionMap
 		return
 	end
 
+	local data
+	if fundFrom ~= nil or discounts ~= nil then
+		data = {
+			tokenCost = tokenCost,
+			fundingPlan = fundFrom and fundingPlan or nil,
+			discounts = discounts,
+		}
```

Comment:
nit - I suggest we introduce a new handler that supports the new object returned by this function. this will give us backwards compatibility as we integrate into arns.app and SDK. 

propose the handler name is `getCostDetailsForAction` or something similar

Reply:
for this refactor, would we leave discounts out of this return object and only put it into the new getCostDetailsForAction return? 

fwiw we can start to gracefully expect object or raw tokenCost on the getTokenCost SDK methods so I'm not super worried about backwards compatibility. but the new handler would make sense so the return is always expected

Reply:
i would keep this to only return the raw token cost, the new handler can return funding plan and disconts

Reply:
77de4b1